common/types.go

```diff
@@ -178,25 +178,57 @@ type RegistryCredential struct {
 	BasicAuth BasicAuthCredential `json:"basicauth,omitempty"`
 }
 
+// Registry describes a template registry
+// TODO(jackr): Fix ambiguity re: whether or not URL has a scheme.
 type Registry struct {
-	Name       string             `json:"name,omitempty"` // Friendly name for the repo
-	Type       RegistryType       `json:"type,omitempty"` // Technology implementing the registry
-	URL        string             `json:"name,omitempty"` // URL to the root of the repo, for example: github.com/helm/charts
+	Name   string         `json:"name,omitempty"`   // Friendly name for the registry
+	Type   RegistryType   `json:"type,omitempty"`   // Technology implementing the registry
+	URL    string         `json:"name,omitempty"`   // URL to the root of the registry
+	Format RegistryFormat `json:"format,omitempty"` // Format of the registry
+}
+
+// AuthenticatedRegistry describes a type registry with credentials.
+type AuthenticatedRegistry struct {
+	Registry
 	Credential RegistryCredential `json:"credential,omitempty"`
-	Format     RegistryFormat     `json:"format,omitempty"`
 }
 
 // RegistryType defines the technology that implements the registry
 type RegistryType string
 
 const (
-	Github RegistryType = "github"
+	GithubRegistryType RegistryType = "github"
 )
 
-// RegistryFormat defines the format of the registry
+// RegistryFormat is a semi-colon delimited string that describes the format
+// of a registry.
 type RegistryFormat string
 
 const (
-	VersionedRegistry   RegistryFormat = "versioned"
+	// Versioning.
+	// If a registry if versioned, then types appear under versions.
+	VersionedRegistry RegistryFormat = "versioned"
+	// If a registry is unversioned, then types appear under their names.
 	UnversionedRegistry RegistryFormat = "unversioned"
+
+	// Organization.
+	// In a collection registry, types are grouped into collections.
+	CollectionRegistry RegistryFormat = "collection"
```

Comment:
What's the expected format for this? It seems like with the addition of these you have two orthogonal "formats" where a registry can be both a collection and versioned registry?

I see the comment above says something about semi-colon delimited, but there is code below that just checks the format against these consts directly.